### PR TITLE
Check that SDK root path exists before versions

### PR DESCRIPTION
**Pull request by nosami**
Fixes VSTS #948292

### CODE REVIEW

File: main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreSdkPaths.cs [hunk 1]
```diff
@@ -190,6 +190,9 @@ bool CheckIsSupportedSdkVersion (string sdkDirectory)
 
 		IEnumerable<DotNetCoreVersion> GetInstalledSdkVersions (string sdkRootPath)
 		{
+			if (!Directory.Exists (sdkRootPath))
+				return Enumerable.Empty<DotNetCoreVersion> ();
+
```
Comment (mrward) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreSdkPaths.cs]: If we are doing a Directory.Exists check here then we do not have do one in the constructor before GetInstalledSdkVersions() is called.

Comment (nosami) [reply main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreSdkPaths.cs]: OK. I will remove the one in the constructor. The check here is needed.

**Comment by nosami** [comment]
@monjenkins backport release-8.3

**Comment by rodrmoya** [comment]
hmm, a related test is failing

**Comment by nosami** [comment]
@rodrmoya I just checked with @mrward. The test is invalid so I removed it.

**Comment by nosami** [comment]
@monojenkins backport release-8.3

**Comment by nosami** [comment]
@monojenkins backport release-8.3

**Comment by nosami** [comment]
@rodrmoya the failing dotnetcore test was due to cached SDK versions persisting in between tests. So the tests could fail depending on the order that they were run in. I guess this could have affected other branches too

**Pull request merged**